src/stores/constraints.ts

```diff
@@ -227,8 +210,24 @@ export const cachedConstraintsStatus: Readable<Status | null> = derived(
   ([$relevantConstraintRuns, $constraintPlanSpecsMap]) => {
     return $relevantConstraintRuns.reduce(
       (status: Status | null, constraintRun: ConstraintRun) => {
+        const constraintPlanSpec =
+          $constraintPlanSpecsMap[constraintRun.constraint_id]?.[constraintRun.constraint_invocation_id];
+        let constraintPlanSpecRevision: number | null = constraintPlanSpec.constraint_revision ?? null;
+        if (constraintPlanSpec.constraint_revision === null) {
+          const constraintVersions = constraintPlanSpec.constraint_metadata?.versions ?? [];
+          constraintPlanSpecRevision = constraintVersions[constraintVersions.length - 1].revision;
+        }
+
+        if (
+          constraintPlanSpec &&
+          (constraintRun.constraint_revision !== constraintPlanSpecRevision ||
+            JSON.stringify(constraintRun.arguments) !== JSON.stringify(constraintPlanSpec.arguments))
+        ) {
+          return Status.Modified;
+        }
+
         if (constraintRun.results.violations?.length) {
-          return Status.Failed;
+          return Status.Complete;
         } else if (status !== Status.Failed) {
           return Status.Complete;
```

Comment:
Not sure if this comment stems from this PR or not but if simulation is out of date, should that be reflected in the rolled-up constraint status? 
<img width="511" alt="image" src="https://github.com/user-attachments/assets/7b777025-6de9-4dfa-b2fd-539bcdedbef3" />

Reply:
Possibly, but the current behavior on dev is that it only reflects the simulation staleness after a new simulation is processed.

Reply:
Cool, no need to change that behavior now then.